### DISCUSSION TITLE
Configure `colcon` for space ROS

### DISCUSSION THREAD

**Discussion by franklinselva** [space-ros/space-ros#371] [Ideas]
The current `colcon` build commands are long and take a bit more effort in development in handling arguments for `cmake`, `ctest`, `ikos` and `pytest`.

### Feature Request

I am proposing the use of `colcon-mixin` to streamline the custom configurations and arguments for different purposes in space-ros. This could be added as a part of the main docker image leveraging a set of sub commands for `colcon` tailored for the needs. The benefits of this feature would be to write policies for building and testing conditions focusing on the standards we want to place. This should lead full control over the development of new applications, simplifying the commands and avoid rewriting the same arguments over and over.

**Comment by EzraBrooks** [comment]
Using `colcon-mixin` and `colcon-defaults` would be a great way to handle this. This is how we handle it on MoveIt Pro - we have a `colcon-defaults.yaml` at the workspace level that contains references to all the mixins and colcon configuration options.

**Comment by Bckempa** [comment]
Was this addressed by https://github.com/space-ros/space-ros/issues/57 or does this still require attention?

**Comment by ivanperez-keera** [comment]
@franklinselva gentle ping.

**Comment by ivanperez-keera** [comment]
We haven't heard anything new on this, so I'm going to move this to a discussion.

Please continue there and, when there are specific changes that we've decided to act upon, we can create issues and PRs.
